tsdownsample/downsamplers.py

```diff
@@ -1,39 +1,93 @@
+import warnings
+from typing import Union
+
+import numpy as np
+
 # ------------------ Rust Downsamplers ------------------
 from tsdownsample._rust import _tsdownsample_rs  # type: ignore[attr-defined]
 
-from .downsampling_interface import RustDownsamplingInterface
+from .downsampling_interface import AbstractDownsampler, AbstractRustDownsampler
 
-MinMaxDownsampler = RustDownsamplingInterface("MinMax", _tsdownsample_rs.minmax)
-M4Downsampler = RustDownsamplingInterface("M4", _tsdownsample_rs.m4)
-LTTBDownsampler = RustDownsamplingInterface("LTTB", _tsdownsample_rs.lttb)
-MinMaxLTTBDownsampler = RustDownsamplingInterface(
-    "MinMaxLTTB", _tsdownsample_rs.minmaxlttb
-)
+rust_dtypes = [
```

Comment:
Will be imported via the `from .downsamplers import *` within the `__init__.py`

Reply:
Valid remark!